bytessource.go

```diff
@@ -0,0 +1,57 @@
+package fuzz
+
+import (
+	"encoding/binary"
+	"math/rand"
+)
+
+// byteSource implements rand.Source determined by a slice of bytes. The random numbers are
+// generated from each 8 bytes in the slice, until the last bytes are consumed, from which a
+// fallback pseudo random source is created in case more random numbers are required.
+//
+// Initialization: create a new byteSource with the input field set to a non empty byte slice. The
+// fallback field should not be set.
+type bytesSource struct {
+	input    []byte
+	fallback rand.Source
+}
+
+func (s *bytesSource) Int63() int64 {
```

Comment:
I recommend implementing Uint64, and then implementing Int63 by calling that and discarding a bit. I think if you implement Uint64(), rand.Rand won't actually call Int63, which is good because it loses a bit :)